thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -41,6 +41,7 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
     Owned<IPartDescriptor> partDesc, tlkDesc;
     IHThorIndexWriteArg *helper;
     Owned <IKeyBuilder> builder;
+    OwnedIFileIO builderIFileIO;
```

Comment:
This will fix the problem, but is the correct fix to add close() to IIOStream, and have builder->finish() call that?
(I am still trying to work out what the correct way of solving this problm is in general.)

Reply:
I don't think IFileIOStream should ever call the IFileIO::close it's working on 
There might be multiple IFileIOStreams using the same IFileIO..

Alternatively, there could be a variety of createKeyBuilder that was parsed a IFileIO, which it could then close, but I don't think that would be quite right either, the create of the IFileIO should close it.

Reply:
I'm not sure it ever makes much sense to have multiple streams writing to a file.  But this change is fine.